standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -378,15 +393,21 @@ public void setConf(Configuration conf){
     int maxPoolSize = MetastoreConf.getIntVar(conf, ConfVars.CONNECTION_POOLING_MAX_CONNECTIONS);
     synchronized (TxnHandler.class) {
       try (DataSourceProvider.DataSourceNameConfigurator configurator =
-               new DataSourceProvider.DataSourceNameConfigurator(conf, "txnhandler")) {
+               new DataSourceProvider.DataSourceNameConfigurator(conf, POOL_TX)) {
         if (connPool == null) {
           connPool = setupJdbcConnectionPool(conf, maxPoolSize);
         }
         if (connPoolMutex == null) {
-          configurator.resetName("mutex");
+          configurator.resetName(POOL_MUTEX);
           connPoolMutex = setupJdbcConnectionPool(conf, maxPoolSize);
         }
       }
+      maxPoolSize = MetastoreConf.getIntVar(conf, ConfVars.HIVE_COMPACTOR_CONNECTION_POOLING_MAX_CONNECTIONS);
+      try (DataSourceProvider.DataSourceNameConfigurator ignored = new DataSourceProvider.DataSourceNameConfigurator(conf, POOL_COMPACTOR)) {
```

Comment:
are we initiating compaction pool even when CompactionTxnHandler is not used?

Reply:
we'll be initiating compaction pool on every HMS instance, however, it should be only done on a leader

Reply:
Fixed